test_conformance/conversions/conversions_data_info.h

```diff
@@ -0,0 +1,812 @@
+//
+// Copyright (c) 2023 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef CONVERSIONS_DATA_INFO_H
+#define CONVERSIONS_DATA_INFO_H
+
+#if defined(__APPLE__)
+#include <OpenCL/opencl.h>
+#else
+#include <CL/opencl.h>
+#endif
+
+#if (defined(__arm__) || defined(__aarch64__)) && defined(__GNUC__)
+#include "fplib.h"
+extern bool qcom_sat;
+extern roundingMode qcom_rm;
+#endif
+
+#include "harness/mt19937.h"
+#include "harness/rounding_mode.h"
+#include "harness/typeWrappers.h"
+
+#include <vector>
+
+#if defined(__linux__)
+#include <sys/param.h>
+#include <libgen.h>
+#endif
+
+extern size_t gTypeSizes[kTypeCount];
+extern void *gIn;
+
+
+typedef enum
+{
+    kUnsaturated = 0,
+    kSaturated,
+
+    kSaturationModeCount
+} SaturationMode;
+
+
+struct DataInitInfo
+{
+    cl_ulong start;
+    cl_uint size;
+    Type outType;
+    Type inType;
+    SaturationMode sat;
+    RoundingMode round;
+    cl_uint threads;
+
+
+    static std::vector<uint32_t> specialValuesUInt;
+    static std::vector<float> specialValuesFloat;
+    static std::vector<double> specialValuesDouble;
+};
+
+
+struct DataInitBase : public DataInitInfo
+{
+    DataInitBase(const DataInitInfo &agg): DataInitInfo(agg) {}
+    virtual void conv_array(void *out, void *in, size_t n) {}
```

Comment:
I don't think it is "needed" technically due to the fact the test constructs/destructs directly `DataInfoSpec`

[    DataInfoSpec<InType, OutType, InFP, OutFP> init_info(info);](https://github.com/shajder/OpenCL-CTS/blob/70838ce82bb70ae3ae8f2f0e889612493661a6d2/test_conformance/conversions/basic_test_conversions.cpp#L550)

Reply:
In the current version it's not strictly needed indeed, but it's good practice to have one if there's a virtual method (to avoid surprises in the future).

Reply:
Done